src/net/tcp/listener.rs

```diff
@@ -306,4 +328,26 @@ cfg_windows! {
             self.watcher.into_inner().unwrap().into_raw_socket()
         }
     }
+
+    cfg_io_safety! {
+        use crate::os::windows::io::{AsSocket, BorrowedSocket, OwnedSocket};
+
+        impl AsSocket for TcpListener {
+            fn as_socket(&self) -> BorrowedSocket<'_> {
+                self.watcher.get_ref().as_socket()
+            }
+        }
+
+        impl From<OwnedSocket> for TcpListener {
+            fn from(fd: OwnedSocket) -> TcpListener {
+                std::net::TcpListener::from(fd).into()
+            }
+        }
+
+        impl From<TcpListener> for OwnedSocket {
+            fn from(listener: TcpListener) -> OwnedSocket {
+                listener.watcher.into_inner().unwrap().into()
```

Comment:
Should this be `TryFrom`?

Reply:
I was mirroring the `IntoRawSocket` implementation for this type. Should this be `TryFrom`? If so, I can make it.